packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.test.js

```diff
@@ -13,139 +13,178 @@
+    expect(add).toHaveBeenCalledWith('id-1');
+    expect(remove).toHaveBeenCalledWith('id-2');
   });
 
   it('sets trace color to red if error tag is present', () => {
-    wrapper.setProps({
-      traces: [
-        {
-          traceID: 'err',
-          traceName: 'T',
-          startTime: 0,
-          duration: 1,
-          processes: {},
-          spans: [{ tags: [{ key: 'error', value: true }] }],
-        },
-      ],
-    });
-    const data = wrapper.find(ScatterPlot).prop('data');
-    expect(data[0].color).toBe('red');
+    const errorTrace = [
+      {
+        traceID: 'err',
+        traceName: 'T',
+        startTime: 0,
+        duration: 1,
+        processes: {},
+        spans: [{ tags: [{ key: 'error', value: true }] }],
+      },
+    ];
+    render(<SearchResults {...baseProps} traces={errorTrace} />);
+    const scatterProps = ScatterPlot.mock.calls[0][0];
+    expect(scatterProps.data[0].color).toBe('red');
   });
 
   it('renders DiffSelection when diffCohort is provided', () => {
-    wrapper.setProps({ diffCohort: [{ id: 'id-1' }] });
-    expect(wrapper.find(DiffSelection)).toHaveLength(1);
+    render(<SearchResults {...baseProps} diffCohort={[{ id: 'id-1' }]} />);
+    expect(screen.getByTestId('diffselection')).toBeInTheDocument();
   });
 
   it('calls goToTrace when a ScatterPlot point is clicked', () => {
-    const goToTrace = jest.fn();
-    const trace = {
-      traceID: 'id-1',
-      spans: [],
-      startTime: 0,
-      duration: 1,
-      traceName: 'TraceThis',
-      processes: {},
-    };
-    wrapper.setProps({ traces: [trace], goToTrace });
-    wrapper.find(ScatterPlot).prop('onValueClick')(trace);
-    expect(goToTrace).toHaveBeenCalledWith('id-1');
+    const goTo = jest.fn();
+    render(<SearchResults {...baseProps} goToTrace={goTo} />);
+    const scatterProps = ScatterPlot.mock.calls[0][0];
+    scatterProps.onValueClick({ traceID: 'a' });
+    expect(goTo).toHaveBeenCalledWith('a');
   });
 
   describe('search finished with results', () => {
     it('shows a scatter plot', () => {
-      expect(wrapper.find(ScatterPlot).length).toBe(1);
+      render(<SearchResults {...baseProps} />);
+      expect(screen.getByTestId('scatterplot')).toBeInTheDocument();
     });
 
     it('shows a result entry for each trace', () => {
-      expect(wrapper.find(ResultItem).length).toBe(traces.length);
+      render(<SearchResults {...baseProps} />);
+      expect(ResultItem.mock.calls).toHaveLength(baseTraces.length);
     });
 
     it('deep links traces', () => {
       const uiFind = 'ui-find';
-      const spanLinks = {
-        [traces[0].traceID]: uiFind,
-      };
-      wrapper.setProps({ spanLinks });
-      const results = wrapper.find(ResultItem);
-      expect(results.at(0).prop('linkTo').search).toBe(`uiFind=${uiFind}`);
-      expect(results.at(1).prop('linkTo').search).toBeUndefined();
+      const spanLinks = { [baseTraces[0].traceID]: uiFind };
```

Comment:
confusing naming
```suggestion
      const spanLinks = { [baseTraces[0].traceID]: 'foobar' };
```
then check for 'foobar' below